client/lib/post-normalizer/rule-content-detect-polls.js

```diff
@@ -22,12 +21,19 @@ export default function detectPolls( post, dom ) {
 		if ( ! noscript.firstChild ) {
 			return;
 		}
-		let firstChildData = decodeEntities( noscript.firstChild.data );
-		let match = firstChildData.match( '^<a href="http:\/\/polldaddy.com\/poll\/([0-9]+)' );
-		if ( match ) {
-			let p = document.createElement( 'p' );
-			p.innerHTML = '<a rel="external" target="_blank" href="http://polldaddy.com/poll/' + match[1] + '">' + i18n.translate( 'Take our poll' ) + '</a>';
-			noscript.parentNode.replaceChild( p, noscript );
+
+		const noscriptDom = document.createElement( 'div' );
+		// some browers don't require this and let us query the dom inside a noscript. some do not. maybe just jsdom.
+		noscriptDom.innerHTML = noscript.innerHTML;
+
+		let pollLink = noscriptDom.querySelector( 'a[href^="http://polldaddy.com/poll/"]' );
+		if ( pollLink ) {
+			const pollId = pollLink.href.match( /http:\/\/polldaddy.com\/poll\/([0-9]+)/ )[ 1 ];
+			if ( pollId ) {
+				let p = document.createElement( 'p' );
+				p.innerHTML = '<a rel="external" target="_blank" href="http://polldaddy.com/poll/' + pollId + '">' + i18n.translate( 'Take our poll' ) + '</a>';
```

Comment:
Fun fact!

Although this probably isn't useful here (don't think it could handle the `real` and `target` attributes) I recently found in the docs that `String.prototype` has a `link` function that generates the link text…

```js
i18n.translate( 'Take our poll' ).link( `http://polldaddy.com/poll/${ pollId }` )
```

^^^ that will produce a string for the `<a/>` tag with the text in the middle :)

who would'a thunk it?